### PR TITLE
Add GCP > CIS v1.3

### LINKED ISSUE #73: Add GCP > CIS v1.3 > 1.16 Ensure Essential Contacts is Configured for Organization
**Is your feature request related to a problem? Please describe.**
Add GCP > CIS v1.3 > 1.16 Ensure Essential Contacts is Configured for Organization

## Description

It is recommended that Essential Contacts is configured to designate email addresses for Google Cloud services to notify of important technical or security information.

Many Google Cloud services, such as Cloud Billing, send out notifications to share important information with Google Cloud users. By default, these notifications are sent to members with certain Identity and Access Management (IAM) roles. With Essential Contacts, you can customize who receives notifications by providing your own list of contacts.

## Remediation

Add essential contacts for the GCP Organization.

### From Console

To add the essential contacts

1. Go to Essential Contacts by visiting https://console.cloud.google.com/iam-admin/essential-contacts
2. Make sure the organization appears in the resource selector at the top of the page. The resource selector tells you what project, folder, or organization you are currently managing contacts for.
3. Click +Add contact
4. In the Email and Confirm Email fields, enter the email address of the contact.
5. From the Notification categories drop-down menu, select the notification categories that you want the contact to receive communications for.
6. Click Save

### LINKED ISSUE #74: Add GCP > CIS v1.3 > 1.17 Ensure that Dataproc Cluster is encrypted using Customer- Managed Encryption Key
**Is your feature request related to a problem? Please describe.**
Add GCP > CIS v1.3 > 1.17 Ensure that Dataproc Cluster is encrypted using Customer- Managed Encryption Key

## Description

When you use Dataproc, cluster and job data is stored on Persistent Disks (PDs) associated with the Compute Engine VMs in your cluster and in a Cloud Storage staging bucket. This PD and bucket data is encrypted using a Google-generated data encryption key (DEK) and key encryption key (KEK). The CMEK feature allows you to create, use, and revoke the key encryption key (KEK). Google still controls the data encryption key (DEK).

Cloud services offer the ability to protect data related to those services using encryption keys managed by the customer within Cloud KMS. These encryption keys are called customer-managed encryption keys (CMEK). When you protect data in Google Cloud services with CMEK, the CMEK key is within your control.

## Remediation

Add essential contacts for the GCP Organization.

### From Console

1. Login to the GCP Console and navigate to the Dataproc Cluster page by visiting https://console.cloud.google.com/dataproc/clusters.
2. Select the project from the projects dropdown list.
3. On the Dataproc Cluster page, click on the Create Cluster to create a new cluster with Customer managed encryption keys.
4. On Create a cluster page, perform below steps:

    * Inside Set up cluster section perform below steps:
        - In the Name textbox, provide a name for your cluster.
            - From Location select the location in which you want to deploy a cluster.
            - Configure other configurations as per your requirements.
    * Inside Configure Nodes and Customize cluster section configure the settings as per your requirements.
    * Inside Manage security section, perform below steps:
        - From Encryption, select Customer-managed key.
        - Select a customer-managed key from dropdown list.
        - Ensure that the selected KMS Key have Cloud KMS CryptoKey Encrypter/Decrypter role assign to Dataproc Cluster service account("serviceAccount:service-<project_number>@compute-system.iam.gserviceaccount.com").
        - Click on Create to create a cluster.
    * Once the cluster is created migrate all your workloads from the older cluster to the new cluster and delete the old cluster by performing the below steps:
        - On the Clusters page, select the old cluster and click on Delete cluster.
        - On the Confirm deletion window, click on Confirm to delete the cluster.
        - Repeat step above for other Dataproc clusters available in the selected project.
    * Change the project from the project dropdown list and repeat the remediation procedure for other Dataproc clusters available in other projects.

### LINKED ISSUE #75: Add GCP > CIS v1.3 > 2.15 Ensure 'Access Approval' is 'Enabled'
**Is your feature request related to a problem? Please describe.**
Add GCP > CIS v1.3 > 2.15 Ensure 'Access Approval' is 'Enabled'

**Description:**
GCP Access Approval enables you to require your organizations' explicit approval whenever Google support try to access your projects. You can then select users within your organization who can approve these requests through giving them a security role in IAM. All access requests display which Google Employee requested them in an email or Pub/Sub message that you can choose to Approve. This adds an additional control and logging of who in your organization approved/denied these requests.

**Audit:**
From Console:

Determine if Access Transparency is Enabled as it is a Dependency
1. From the Google Cloud Home inside the project you wish to audit, click on the Navigation hamburger menu in the top left. Hoverover the IAM & Admin Menu. Select settings in the middle of the column that opens.
2. The status should be "Enabled' under the heading Access Transparency 

Determine if Access Approval is Enabled:

1. From the Google Cloud Home, within the project you wish to check, click on the Navigation hamburger menu in the top left. Hover over the Security Menu. Select Access Approval in the middle of the column that opens.
2. The status will be displayed here. If you see a screen saying you need to enroll in Access Approval, it is not enabled.

From CLI:

Determine if Access Approval is Enabled
1. From within the project you wish to audit, run the following command.
`gcloud access-approval settings get`
2. The status will be displayed in the output.

### PR DESCRIPTION
**Pull request by vkumbha**
Closes #75, Closes #74, Closes #73

### CODE REVIEW

**Review by rajlearner17 — APPROVED** [review]
As per testing of new controls

File: cis_v130/docs/cis_v130_3_10.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+- Source IP ranges (press Enter after you paste each value in the box, copy the value below the bold text including the dash):
+
+    IAP Proxy Addresses
+    - 35.235.240.0/20
+
+    Google Health Check
+    - 130.211.0.0/22
+    
+      **```Google Health Check
```
Comment (misraved) [inline cis_v130/docs/cis_v130_3_10.md:30]:
```suggestion
      **Google Health Check
```

File: cis_v130/docs/cis_v130_3_4.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  --dnssec-state on \
+  --ksk-algorithm ECDSAP256SHA256 \
+  --ksk-key-length 256 \
+  --zsk-algorithm ECDSAP256SHA256 \
+  --zsk-key-length 256 \
+  --denial-of-existence NSEC
+```
+**Note** Supported algorithm options and key lengths are as follows.
+
+```
```
Comment (misraved) [inline cis_v130/docs/cis_v130_3_4.md:48]:
```suggestion
```bash
```

File: cis_v130/docs/cis_v130_3_5.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  --dnssec-state on \
+  --ksk-algorithm ECDSAP256SHA256 \
+  --ksk-key-length 256 \
+  --zsk-algorithm ECDSAP256SHA256 \
+  --zsk-key-length 256 \
+  --denial-of-existence NSEC
+```
+**Note** Supported algorithm options and key lengths are as follows.
+
+```
```
Comment (misraved) [inline cis_v130/docs/cis_v130_3_5.md:48]:
```suggestion
```bash
```

File: query/organization/organization_essential_contacts_configured.sql [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    when to_jsonb(array_agg(notificationtype)) @> '["ALL"]'::jsonb
+      then title || ' essential contacts are configured.'
+    else title || ' essential contacts are not configured.'
+  end reason,
+  -- Additional Dimensions
+  organization_id
+from
+  categories
+group by
+  name, title, organization_id;
```
Comment (misraved) [inline query/organization/organization_essential_contacts_configured.sql:33]:
```suggestion
  name, 
  title, 
  organization_id;
```

File: query/dataproc/dataproc_cluster_encryption_with_cmek.sql [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    when config -> 'encryptionConfig' ->> 'gcePdKmsKeyName' is null then 'alarm'
+    else 'ok'
+  end status,
+  case
+    when config -> 'encryptionConfig' ->> 'gcePdKmsKeyName' is null
+      then title || ' is not encrypted using customer-managed encryption keys (CMEK).'
+    else title || ' is encrypted using customer-managed encryption keys (CMEK).'
+  end reason,
+  -- Additional Dimensions
+  project
```
Comment (misraved) [inline query/dataproc/dataproc_cluster_encryption_with_cmek.sql:14]:
```suggestion 
  location,
  project
```

**Review by rajlearner17 — APPROVED** [review]

**Pull request merged**